### DISCUSSION TITLE
New collection review: inspur.ispim

### DISCUSSION THREAD

**Discussion by ispim** [ansible-collections/ansible-inclusion#47] [Resolved reviews]
Hello, I'd like to submit the inspur.ispim collection for review:

Ansible Galaxy: https://galaxy.ansible.com/inspur/ispim
Github repo: https://github.com/ispim/inspur.ispim
Github handles: @ispim @baoshanwang

**Comment by ispim** [comment]
@Andersson007 Could you please help to check whether our new collection meets the requirements?

**Reply by Andersson007** [reply]
@ispim hello, thank you for submitting a new collection! i'm very busy with other tasks now. When there's a chance, I'll take a look of course.
Maybe someone from the @ansible-collections/steering-committee can help with the first review?

**Comment by ispim** [comment]
@ansible-collections/steering-committee Could you please help to check whether our collection meets the requirements?

**Comment by Andersson007** [comment]
# Ansible Collections Checklist (short version)

_For details about the following points, refer to the [Collection Requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst)._

Every comment should say whether the reviewer expects it to be addressed, or whether it's optional.

**Public availability and communication:**
- [x] published on [Ansible Galaxy](https://galaxy.ansible.com) with version 1.0.0 or later
- [x] has a policy of releasing, versioning and deprecation announced to contributors and users in some way
- [x] have a Code of Conduct (CoC) compatible with the [Ansible Code of Conduct (CoC)](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
- [x] have a publicly available issue tracker that does not require a paid level of service to create an account or view issues
- [x] have a public SCM repository, releases of the collection are tagged in this repository

**Standards and documentation:**
- [x] adheres to [semantic versioning](https://semver.org/)
- [x] follows [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
- [x] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
  **MUST FIX**: Please add the `notes:` section to every module with `- Supports C(check_mode).` or `- Does not support C(check_mode).`
  **SHOULD FIX**: If you look at README, you can notice several formatting issues.
- [x] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html); as well as these other requirements:
  - [x] modules that only gather information are named `<something>_info`
  - [x] modules that return `ansible_facts` are named `<something>_facts` and do not return non-facts
  - [x] other modules must not allow querying information using specific `state` option values, or similar mechanisms (like `state=get` or `state=query`).  These features should be moved to `<something>_info` or `<something>_fact` modules.
  - [x] `check_mode` is supported in all `_info` and `_facts` modules
- [x] supports Python 2.6 or greater and Python 3.5 or greater. If it does not, read the [full guidelines](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#python-compatibility) to see if you qualify for an exception and document the unsupported [Python versions](https://docs.ansible.com/ansible/latest/dev_guide/developing_python_3.html#ansible-and-python-3) in the collection ``README.md`` and in every module and plugin (or in doc fragments)
  **MUST FIX**: Please add the `requirements:` section with `ism` and Python 3.7+ to the doc fragment
- [x] only uses the [allowed plugin types](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#modules-plugins) in the `plugins/` directory
- [x] has `README.md`
- [x] documentation, examples, and return sections use FQCNs for the `M(..)` [format macros](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#linking-and-other-format-macros-within-module-documentation) when referring to modules
- [x] modules (or plugins) from ansible-core use `ansible.builtin.` as a FQCN prefix
- [x] documentation and return sections use `version_added:` containing the *collection* version for which an option, module or plugin was added
- [x] FQCNs are used for `extends_documentation_fragment:`, unless you are referring to doc_fragments from ansible-core
- [x] public plugins, roles and playbooks do not use files outside of `meta/`, `plugins/`, `roles/`, and `playbooks/`

**Collection management:**
- [x] `galaxy.yml` having `tags` field set
- [x] collection dependencies must have a lower bound on the version which is at least 1.0.0, and are all part of the `ansible` package
- [x] `meta/runtime.yml` defines the minimal version of Ansible which the collection works with
  **MUST FIX**: in README is said that the collection requires Ansible 2.10 but in `meta/runtime.yml` is written `requires_ansible: '>=2.9.10'`.
- [x] has changelog, preferably with `changelogs/changelog.yaml`
- [x] collection repository should not contain any large objects (binaries) comparatively to the current Galaxy tarball size limit of 20 MB. For example, package installers for testing purposes shouldn't be added.
- [x] collection repository should not contain any unnecessary files like, for example, temporary files created by development tools. Temporary files MUST be added to `.gitignore`.
  **COMMENT:** is the `bindep.txt` file really needed?

**Tests:**

- [x] passed `ansible-test sanity`
- [x] if `test/sanity/ignore*.txt` exists, it does not contain error codes listed [here](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#ci-testing)
- [ ] has CI tests up and running against each of the "major versions" of `ansible-base`/`ansible-core` that the collection supports
  **MUST FIX:** Please add `stable-2.13` to your test matrix
- [x] all CI tests run against every pull request
- [x] all CI tests run against a commit that releases the collection; if they don't pass, the collection won't be released
- [x] all CI tests run regularly (nightly, or at least once per week)

**MUST FIX** If you and other collection maintainers are not subscribed to the [news-for-maintainers](https://github.com/ansible-collections/news-for-maintainers) repo, please do it not to miss important CI-related updates.

**Reply by Andersson007** [reply]
@ispim i've finished the first round of review ^, could you please take a look and fix `MUST FIX`es and comment on `COMMENT`s/`SHOULD FIX`es?
Thanks!

**Reply by ispim** [reply]
Thank you very much. We will start modifying it right away.

**Reply by ispim** [reply]
@Andersson007 We have modified the problems found. Please check whether they meet the requirements.

**Reply by Andersson007** [reply]
@ispim it's not an obstacle, just a fyi about README formatting issues:

![1](https://user-images.githubusercontent.com/34477873/174242412-13f0b703-f01d-4fb4-88ce-54a819db069e.PNG)

**Reply by Andersson007** [reply]
@ispim anyway MUST FIXes LGTM now, so +1 from me for inclusion. I'll change the category that the second review is needed, thanks!

**Comment by ispim** [comment]
Hello, may I ask when the second system audit will start? Will it be too late to add our collection to Ansible 6 version?

**Reply by Andersson007** [reply]
Hello @ispim ,

>may I ask when the second system audit will start?

I post all the collections waiting for review in [Bullhorn](https://github.com/ansible/community/wiki/News#the-bullhorn) every week.
When someone from the Community or Steering Committee decides to conduct the second review, then it'll start. The participation is volunteer.

>Will it be too late to add our collection to Ansible 6 version?

No, it won't be late as we can include it in a minor release

**Reply by ispim** [reply]
Ok, thank you very much.

**Comment by Andersson007** [comment]
@ansible-collections/steering-committee could anyone please do the second review?

**Comment by ispim** [comment]
@ansible-collections/steering-committee When can the second review begin?

**Reply by Andersson007** [reply]
@ispim when there's a volunteer.
I publish a list of the collections that are waiting for inclusion reviews asking for help every week in [Bullhorn](https://github.com/ansible/community/wiki/News#the-bullhorn) (though haven't done it during the previous two weeks as was on PTO).
Hope someone will respond ASAP.

**Comment by mariolenz** [comment]
# Ansible Collections Checklist (short version)

_For details about the following points, refer to the [Collection Requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst)._

Every comment should say whether the reviewer expects it to be addressed, or whether it's optional.

Note for reviewers: If you don't know how to check any of the points below, please ask maintainers of the collection you're reviewing or a [Steering Committee member](https://docs.ansible.com/ansible/devel/community/steering/community_steering_committee.html#current-steering-committee-members) for clarifications in comments of corresponding inclusion discussion.

**Public availability and communication:**
- [x] published on [Ansible Galaxy](https://galaxy.ansible.com) with version 1.0.0 or later
- [x] has a policy of releasing, versioning and deprecation announced to contributors and users in some way
- [x] have a Code of Conduct (CoC) compatible with the [Ansible Code of Conduct (CoC)](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
- [x] have a publicly available issue tracker that does not require a paid level of service to create an account or view issues
- [x] have a public SCM repository, releases of the collection are tagged in this repository

**Standards and documentation:**
- [x] adheres to [semantic versioning](https://semver.org/)
- [x] follows [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
- [x] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
- [x] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html); as well as these other requirements:
  - [x] modules that only gather information are named `<something>_info`
  - [x] modules that return `ansible_facts` are named `<something>_facts` and do not return non-facts
  - [x] other modules must not allow querying information using specific `state` option values, or similar mechanisms (like `state=get` or `state=query`).  These features should be moved to `<something>_info` or `<something>_fact` modules.
  - [x] `check_mode` is supported in all `_info` and `_facts` modules
- [x] supports Python 2.6 or greater and Python 3.5 or greater. If it does not, read the [full guidelines](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#python-compatibility) to see if you qualify for an exception and document the unsupported [Python versions](https://docs.ansible.com/ansible/latest/dev_guide/developing_python_3.html#ansible-and-python-3) in the collection ``README.md`` and in every module and plugin (or in doc fragments)
- [x] only uses the [allowed plugin types](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#modules-plugins) in the `plugins/` directory
- [x] has `README.md`
- [x] documentation, examples, and return sections use FQCNs for the `M(..)` [format macros](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#linking-and-other-format-macros-within-module-documentation) when referring to modules
- [x] modules (or plugins) from ansible-core use `ansible.builtin.` as a FQCN prefix
- [x] documentation and return sections use `version_added:` containing the *collection* version for which an option, module or plugin was added
- [x] FQCNs are used for `extends_documentation_fragment:`, unless you are referring to doc_fragments from ansible-core
- [x] public plugins, roles and playbooks do not use files outside of `meta/`, `plugins/`, `roles/`, and `playbooks/`

**Collection management:**
- [x] `galaxy.yml` having `tags` field set
- [x] collection dependencies must have a lower bound on the version which is at least 1.0.0, and are all part of the `ansible` package
- [x] `meta/runtime.yml` defines the minimal version of Ansible which the collection works with
- [x] has changelog, preferably with `changelogs/changelog.yaml`
- [x] collection repository should not contain any large objects (binaries) comparatively to the current Galaxy tarball size limit of 20 MB. For example, package installers for testing purposes shouldn't be added.
- [x] collection repository should not contain any unnecessary files like, for example, temporary files created by development tools. Temporary files MUST be added to `.gitignore`.

**Tests:**

Note for reviewers: If you don't know how to check the points below, please ask maintainers of the collection you're reviewing how you can do it.
* In most cases, CI is set up via GitHub Actions.
* Check `.yml` files in the `.github/workflows` directory. There must be at least the `sanity` section under `jobs` containing the `ansible-test sanity` command running against all supported ansible-core versions that must be also listed there, for example, `- stable-2.11, - stable-2.12, - stable-2.13`.
* Check workflow runs by clicking the `Actions` tab in the repository's page - you're interested in `Scheduled` runs, runs against release commits and runs against pull requests.
* If there are no workflows in the `Actions` tab, ask the collection maintainers how CI is implemented.

- [x] passed `ansible-test sanity`
- [x] if `test/sanity/ignore*.txt` exists, it MUST not contain error codes listed [here](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#ci-testing)
- [x] has CI tests up and running against each of the "major versions" of `ansible-base`/`ansible-core` that the collection supports
- [x] all CI tests MUST run against every pull request
- [x] all CI tests MUST run regularly (nightly, or at least once per week)
- [x] sanity tests MUST run against a commit that releases the collection; if they don't pass, the collection won't be released

**Comment by mariolenz** [comment]
@ispim I think you effectively don't test agains stable. It looks like the CI tests against stable are canceled because the one against devel fails with `Package 'ansible-core' requires a different Python: 3.8.13 not in '>=3.9'` since July 20. But I'm not 100% sure if that is really the root cause.

Apart from this LGTM.

cc @Andersson007

**Reply by Andersson007** [reply]
@mariolenz thanks much for your review!

**Comment by Andersson007** [comment]
@ispim please take into consideration mariolenz's [comment](https://github.com/ansible-collections/ansible-inclusion/discussions/47#discussioncomment-3455961).

I've changed the status to "In process of inclusion". We'll create a corresponding community topic and will set up a default day when the collection will be included and if there are no objections, we'll do it.

@ispim could you please:
- Subscribe to the [news-for-maintainers](https://github.com/ansible-collections/[news-for-maintainers](https://github.com/ansible-collections/news-for-maintainers)) repo and track announcements there. The ansible-core guys publish changes related to CI (and not only). Otherwise, your CI will get red one day in PRs for unclear reasons.
- Optionally but highly recommended, subscribe to [Bullhorn](https://github.com/ansible/community/wiki/News#the-bullhorn).
- We have the [Maintainer guidelines](https://github.com/ansible/community-docs/blob/main/maintaining.rst). Might be useful to take a look.

Collections with red CI are considered unmaintained and are removed from the package from time to time, so be sure you and other maintainers are subscribed at least to news-for-maintainers and track it, thanks!

**Comment by ispim** [comment]
Thank you very much for your advice. I have subscribed to the related project. As for the content of CI test, we have modified it. Please check it. Thank you.

**Comment by mariolenz** [comment]
@ispim ansible-community/ansible-build-data#158 has been merged.
The collection will appear in the package in the next 6.X minor release (which happens every 3 weeks).
Thanks everyone!
